src/parallel.cpp

```diff
@@ -617,54 +626,51 @@ void HandleBlockMessageThread(CNode *pfrom, const string strCommand, const CBloc
 }
 
 
-CCheckQueue<CScriptCheck> *CAllScriptCheckQueues::GetScriptCheckQueue()
+// for newly mined block validation, return the first queue not in use.
+CCheckQueue<CScriptCheck> *CParallelValidation::GetScriptCheckQueue()
 {
-    // for newly mined block validation, return the first queue not in use.
-    CCheckQueue<CScriptCheck> *pqueue = NULL;
-    if (Size() > 0)
+    while (true)
     {
```

Comment:
Yeah; I hesitated here.  There was an if statement that was pointless, and if it failed we asserted.  This was too ugly / dangerous and I chose to just get rid of it.  The consequent re-indent was unfortunate indeed.